utils/remote_control/navigator_keyboard_control/nodes/navigator_keyboard_server.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+
+'''
+Keyboard Server: The keyboard server enables a client to control NaviGator
+using a remote keyboard. It runs a service that responds to KeyboardControl
+service requests. It manages the currently locked client through the use of
+UUIDs. Only keys received from the client with the locked UUID will be
+executed.
+'''
+
+
+import curses
+import uuid
+
+from navigator_msgs.srv import KeyboardControl
+from remote_control_lib import RemoteControl
+import rospy
+
+
+__author__ = "Anthony Olive"
+__maintainer__ = "Anthony Olive"
+__email__ = "anthony@iris-systems.net"
+__copyright__ = "Copyright 2016, MIL"
+__license__ = "MIT"
+
+
+rospy.init_node('keyboard_server')
+
+
+class KeyboardServer(object):
+
+    def __init__(self):
+        self.force_scale = rospy.get_param("/joystick_wrench/force_scale", 600)
+        self.torque_scale = rospy.get_param("/joystick_wrench/torque_scale", 500)
+
+        self.remote = RemoteControl('keyboard', "/wrench/keyboard")
+        rospy.Service('/keyboard_control', KeyboardControl, self.key_recieved)
+
+        # Initialize this to a random UUID so that a client without a UUID cannot authenticate
+        self.locked_uuid = uuid.uuid4().hex
+
+    def key_recieved(self, req):
+        '''
+        This function handles the process of locking control of the service to
+        one client. If an 'L' is received, a UUID is generated for the client
+        and control of the service is locked to it.
+        '''
+
+        # If the key pressed was L, locks control of the service to the clinet's UUID
+        if (req.keycode == 76):
+
+            # Generates a new UUID for the client if it does not already have one
+            if (req.uuid is ''):
+                self.locked_uuid = uuid.uuid4().hex
+            else:
+                self.locked_uuid = req.uuid
+            return {"generated_uuid": self.locked_uuid, "is_locked": True}
+
+        # If the key was from the client with locked control, pass it to execution
+        elif (req.uuid == self.locked_uuid):
+            self.execute_key(req.keycode)
+            return {"generated_uuid": '', "is_locked": True}
+
+        # Ignore keys sent by a client that has not locked control of the service
+        else:
+            return {"generated_uuid": '', "is_locked": False}
+
+    def execute_key(self, key):
```

Comment:
As I always say, there's almost always a "cleaner" way to do things rather than using a lot of `elif`'s.
For example, you could have a dictionary that maps key's to functions (`self.handle_key = {ord('k'): self.remote.toggle_kill, ...}`). That way this function would be one line: `self.handle_key[key]()`.

Not asking for a change just suggesting